### PR TITLE
Typography updates 📚

### LINKED ISSUE #51: Typography updates 🔠
## Summary

Screenshot 2019-08-01 at 16 24 51

New **typographical specs** for our brand.

These updates are meant to be ship as part of the new brand in the [`2.0` milestone](https://github.com/zopaUK/react-components/milestone/1).

## Proposal

I suggested we cover all our typographical needs with two components:
```jsx
/**
 * s = 20px
 * m = 24px [default]
 * l = 28px
 * xl = 48px
 * 
 * line-height: 1.4 
 * weight: 600
 */
<Heading size={'s'|'m'|'l'|'xl'} />

/**
 * s = 12px
 * m = 14px [default]
 * l = 16px
 * 
 * line-height: 1.4 
 */
<Text size={'s'|'m'|'l'} bold={ true | false } />
```
They should cover most type scenarios leveraging the [`as` prop](https://github.com/zopaUK/react-components/issues/24).

## Related issues

- #22 
- #24

### PR DESCRIPTION
**Pull request by lluia**
## Summary 🌮

Apply **typographic updates** from the new brand in the library.

This PR is composed of 3 main changes:

- Remove **all components** under `src/components/typography`
- Use only two components to render text: `<Text />` and `<Heading />`
- Standardise typography rules across **all other components**

## Design spec 📐

62302000-de972280-b479-11e9-93c2-074aa6f5a8ba

## API 💍

Before we were using many components to render text:
- `<Paragraph />`
- `<Subhead`
- _etc..._

Now only...

#### `<Text />` 

<details><summary>Summary 💅🏻</summary>
<p>

- Has three size variations as per the spec: `"large" | "medium" (default) | "small" `
- Renders a `span` by default but you can make it render a `p` through the `as`
- You can set it to have _bottom margin_ through the `mb` prop
- Rendering different HTML tags don't affect the style, is just HTML semantics
- You can control its weight through the `weight` prop
- Its `line-height` is dynamic: **1.6**

</p>
</details>

#### `<Heading />`

<details><summary>Summary 💅🏻</summary>
<p>

- You can control the tag it renders through the `as` props
- Has fours size variations tied to the tag they render: `h1 | h2 | h3 | h4`
- It always renders as `bold`
- Its `line-height` is dynamic: **1.6**

</p>
</details>

## Context 👓

Fixes #51

### CODE REVIEW

**Comment by lluia** [comment]
**Highly suggested:** _review by commit_ and try the changes locally 💅🏻

**Comment by lluia** [comment]
### Todo

- [ ] update documentation on `<Heding />`

File: src/components/atoms/Button/Button.tsx [hunk 1]
```diff
@@ -102,10 +102,10 @@ const SButton = styled.button<IButtonProps>`
   align-items: center;
   width: ${props => props.fullWidth && '100%'};
   padding: ${({ sizing }) => sizing && paddings[sizing]};
-  font-family: ${openSans};
+  font-family: ${typography.primary};
   font-size: ${({ sizing }) => sizing && fontSizes[sizing]};
   line-height: ${({ sizing }) => (sizing === 'large' ? '30px' : '24px')};
-  font-weight: 500;
+  font-family: ${typography.weights.semibold};
```
Comment (rominamoya) [inline src/components/atoms/Button/Button.tsx]: @lluia should this be font-weight?

Comment (lluia) [reply src/components/atoms/Button/Button.tsx]: Ahhh you catch 💯 , yez should be `font-weight` 👍

File: src/components/atoms/Badge/Badge.md [hunk 2]
```diff
@@ -4,26 +4,38 @@ There's four different styles available:
 
 ```js
 import { Fragment } from 'react';
-import { Badge } from '@zopauk/react-components';
+import { Badge, SizedContainer } from '@zopauk/react-components';
```
Comment (lluia) [inline src/components/atoms/Badge/Badge.md]: I need to remove `SizedContainer` from here as it's not used anymore on the example 💅🏻

File: src/components/atoms/Heading/Heading.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+Sizes:
```
Comment (lluia) [inline src/components/atoms/Heading/Heading.md]: Missing examples and prop documentation 🧐

File: tsconfig.json [hunk 4]
```diff
@@ -15,7 +15,8 @@
     "forceConsistentCasingInFileNames": true,
     "resolveJsonModule": true,
     "isolatedModules": true,
-    "noEmit": true
+    "noEmit": true,
+    "removeComments": false
```
Comment (lluia) [inline tsconfig.json]: I need to remove `removeComments`; I used it for debugging...

File: src/components/styles/Fonts.tsx [hunk 5]
```diff
@@ -1,29 +1,6 @@
...
-         /* Pretty Modern Browsers */
-         url('https://s3-eu-west-1.amazonaws.com/cdn.zopa.com/fonts/1333416/99d874d5-3f1c-419c-9c82-4c63e62f98ca.woff')
-         format('woff'),
-         /* Safari, Android, iOS */
-         url('https://s3-eu-west-1.amazonaws.com/cdn.zopa.com/fonts/1333416/c41ae233-742a-4c14-af50-6b0e62e12a58.ttf')
-         format('truetype');
-  }
-
-  /* Open Sans */
   @import url('https://fonts.googleapis.com/css?family=Open+Sans:300,400,600,700&display=swap');
```
Comment (lluia) [inline src/components/styles/Fonts.tsx]: We're only using **three weights** now, I need to update this `@import` accordingly 💅🏻

File: src/components/atoms/Heading/Heading.tsx [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+const {
+  sizes: { heading: headingSizes },
+} = typography;
+
+interface IStyledHeadingProps extends HTMLAttributes<HTMLHeadingElement> {
+  as: 'h1' | 'h2' | 'h3' | 'h4';
+}
+
+const StyledHeading = styled(Text)<IStyledHeadingProps>`
+  font-size: ${({ as = 'h3' }) => headingSizes[as]};
```
Comment (thegrinder) [inline src/components/atoms/Heading/Heading.tsx]: I'm not sure if you need to give heading a default. When you write regular HTML, you still have to pick between h1 and h6

File: src/components/atoms/Heading/Heading.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  letter-spacing: -0.5px;
+`;
```
Comment (thegrinder) [inline src/components/atoms/Heading/Heading.tsx]: no line-height set here? and no bottom margin like in `<Text />`?

Comment (lluia) [reply src/components/atoms/Heading/Heading.tsx]: The headings have a default `margin` applied.

I believe it comes from the browser as [`nomarlize.css`](https://necolas.github.io/normalize.css/8.0.1/normalize.css) doesn't override the default styles for headings 🤔

Happy to set it to `margin: 0` and have a `mb` prop instead 👍🏻

Comment (thegrinder) [reply src/components/atoms/Heading/Heading.tsx]: I think that would be more predictable than relying on browser defaults. What line heights are we using?

Comment (lluia) [reply src/components/atoms/Heading/Heading.tsx]: My idea was to set a [global `line-height`](https://github.com/zopaUK/react-components/blob/9ed9c7ba496f5a053fd59942e2d7356d38958d2d/src/constants/typography.ts#L24), however, I noticed I forgot 🤦🏻‍♂️

For now, I just set a relative value, [`1.6`](https://github.com/zopaUK/react-components/blob/9ed9c7ba496f5a053fd59942e2d7356d38958d2d/src/constants/typography.ts#L24),  happy to know if there's a better approach though 👍🏻

For instance, for buttons, I didn't change the `font-size` and `line-height` as it seemed tricky to align them right 😓

Comment (thegrinder) [reply src/components/atoms/Heading/Heading.tsx]: That would be a possible solution. However, one scenario where I see it getting problematic is if a project has a global line-height already set. If you have it in the component, it's usually enough to overwrite global line-height which allows you to gradually introduce the components without having to mess with global styles

**Pull request merged**